src/js/components/agency/visualizations/StatusOfFundsChart.jsx

```diff
@@ -554,21 +624,66 @@ const StatusOfFundsChart = ({
                 if (window.innerWidth >= 992 && window.innerWidth < 1200) {
                     return 'translate(-150,-85)';
-                } else if (isMediumScreen && !isMobile) {
+                }
+                else if (isMediumScreen && !isMobile) {
                     return 'translate(-150,-90)';
-                } else if (!isLargeScreen) {
+                }
+                else if (!isLargeScreen) {
                     return 'translate(60,0)';
                 }
                 return 'translate(-150,-135)';
             };
 
-            // scale to x and y data points
-            if (sortedNums[sortedNums.length - 1]._outlays < 0) {
-                x.domain(d3.extent(sortedNums, (d) => d._outlays)).nice(2);
+            // scale to x data points;
+            // if there are negative values, use the largest absolute value of the negative values in domain;
+            // if not, use the largest value;
+            // and we have to use separate arrays for the values because outlays has to be separate because they are not always shown
+            const negativeTbrArray = [];
+            const positiveTbrArray = [];
+            const negativeOutlaysArray = [];
+            const positiveOutlaysArray = [];
+
+            sortedNums.forEach((item) => {
+                if (item._outlays < 0) {
+                    negativeOutlaysArray.push(item._outlays);
+                    setNegativeOutlay(true);
+                }
+                if (item._outlays >= 0) {
+                    positiveOutlaysArray.push(item._outlays);
+                }
+                if (item._budgetaryResources < 0) {
+                    negativeTbrArray.push(item._budgetaryResources);
+                    setNegativeTbr(true);
+                }
+                if (item._budgetaryResources >= 0) {
+                    positiveTbrArray.push(item._budgetaryResources);
+                }
+            });
+
+            const maxPosTbr = positiveTbrArray.length ? positiveTbrArray.reduce((a, b) => Math.max(a, b)) : null;
+            const maxNegTbr = negativeTbrArray.length ? negativeTbrArray.reduce((a, b) => Math.max(Math.abs(a), Math.abs(b))) : null;
+            const maxPosOutlay = positiveOutlaysArray.length ? positiveOutlaysArray.reduce((a, b) => Math.max(a, b)) : null;
+            const maxNegOutlay = negativeOutlaysArray.length ? negativeOutlaysArray.reduce((a, b) => Math.max(Math.abs(a), Math.abs(b))) : null;
+
+            const arrayOfMaxValues = [];
+            if (negativeTbr) {
+                arrayOfMaxValues.push(maxNegTbr);
+            }
+            else arrayOfMaxValues.push(maxPosTbr);
+            if (negativeOutlay) {
+                arrayOfMaxValues.push(maxNegOutlay);
+            }
+            else {
+                arrayOfMaxValues.push(maxPosOutlay);
+            }
+
+            if (negativeTbr || negativeOutlay) {
+                x.domain(d3.extent(arrayOfMaxValues)).nice(2);
```

Comment:
could this code be put in a function since it's a duplicate of code above.  just pass in a parameter to determine if the code is meant  for outlay for obligation

Reply:
I tried several ways to condense the code and make it less repetitive.
This code could be in a function.
I didn't do it that way because I wasn't certain the gain would be worth the time to refactor it.